### ISSUE TITLE
Sankey chart ignores edges with source or target null

### ISSUE THREAD

**Issue by peterviergutz** [apache/incubator-echarts#11637]
When trying to add edges with no connection to either a previous or following node, the value of the edge is not added to the existing node.

### Version
4.2.1

### Reproduction link
[https://jsfiddle.net/s2p84c1o/1/](https://jsfiddle.net/s2p84c1o/1/)

### Steps to reproduce
```js
var myChart = echarts.init(document.getElementById('main'));

var option = {
  tooltip: {
    trigger: 'item',
    triggerOn: 'mousemove'
  },
  series: {
    type: 'sankey',
    focusNodeAdjacency: 'allEdges',
    data: [{
      name: 'a'
    }, {
      name: 'b'
    }, {
      name: 'a1'
    }, {
      name: 'a2'
    }, {
      name: 'b1'
    }, {
      name: 'c'
    }],
    links: [{
      source: 'a',
      target: null, // <-- this node does not have a connection
      value: 5
    }, {
      source: 'a',
      target: 'a2',
      value: 3
    }, {
      source: 'b',
      target: 'b1',
      value: 8
    }, {
      source: 'a',
      target: 'b1',
      value: 3
    }, {
      source: 'b1',
      target: 'a1',
      value: 1
    }, {
      source: 'b1',
      target: 'c',
      value: 2
    }]
  }
};

myChart.setOption(option);
```

### What is expected?
Node `a` should have value 11 (5 + 3 + 3) with links to `b1` with value 3 and `a2` with value 3 and no link to target for 5/11 height of the node. 

### What is actually happening?
Edge without target is ignored and not added to node `a` in depth 0. Node `a` has value 6 with links to `b1` and `a2`

**Comment by peterviergutz** [comment]
Dug a bit on this one, the reason is that edges with no target or source are ignored here: 

https://github.com/apache/incubator-echarts/blob/1bdb91aec185937ade601ccb58da13d382fc340d/src/data/Graph.js#L160

and node size is calculated from incoming and outgoing edges here:

https://github.com/apache/incubator-echarts/blob/fcfa95012ca561becab607b065a62b5f87220801/src/chart/sankey/sankeyLayout.js#L86

A quick fix might involve summing the values of nodes with no connection in the latter function like this 

```js
function computeNodeValues(nodes, seriesModel) {

  var nodeSizes = {}
  seriesModel.option.links.filter(link => {
      return !link.source || !link.target
  }).reduce(function (nodeSizes, node) {
      if(!nodeSizes[node.source]) nodeSizes[node.source] = 0
      nodeSizes[node.source] += node.value
      return nodeSizes
  }, nodeSizes)

  zrUtil.each(nodes, function (node) {
    var extra = nodeSizes[node.id] || 0
    var value1 = sum(node.outEdges, getEdgeValue);
    var value2 = sum(node.inEdges, getEdgeValue);
    var value = Math.max(value1, value2);
    node.setLayout({
      value: value + extra
    }, true);
  });
}
```

**Comment by deqingli** [comment]
This is the strategy we use, I want to know why you need to add such an edge with no connection in your scene.

**Comment by peterviergutz** [comment]
We want to offer a flow chart that an overview how many of our users moved between the different stages and buckets over time:

![expected](https://user-images.githubusercontent.com/568682/68852007-4bf05b80-06d7-11ea-9b0f-f18b10120181.png)

Because only a few user "move" the current implementation skews the picture towards a heavy movement of users:

![actual](https://user-images.githubusercontent.com/568682/68852016-53176980-06d7-11ea-97bb-a864a1564c93.png)

After looking into the code; I would rephrase the issue as _we need a way to define the size of a node_, implementing this with "nodes without a connection" is not the way to achieve this. Maybe an optional new property `size` on the nodes' properties would be a cleaner way to do this:

```json
{
  "nodes": [
    {
      "name": "a1",
      "size": 100
    }
...
```

To get the node's size in the chart could then be determined by whatever is bigger: the sum of incoming and outgoing edges or the node's defined size. 

https://github.com/apache/incubator-echarts/blob/fcfa95012ca561becab607b065a62b5f87220801/src/chart/sankey/sankeyLayout.js#L86

**Comment by peterviergutz** [comment]
created a feature request here https://github.com/apache/incubator-echarts/issues/11644
